packages/firestore/src/platform/browser/webchannel_connection.ts

```diff
@@ -356,11 +360,32 @@ export class WebChannelConnection extends RestConnection {
       }
     );
 
+    let isLongPollingConnection:boolean|null = null;
+    let hasRaisedTimeToFirstByteEvent = false;
+
     unguardedEventListen<StatEvent>(requestStats, Event.STAT_EVENT, event => {
       if (event.stat === Stat.PROXY) {
         logDebug(LOG_TAG, 'Detected buffering proxy');
+        isLongPollingConnection = true;
       } else if (event.stat === Stat.NOPROXY) {
         logDebug(LOG_TAG, 'Detected no buffering proxy');
+        isLongPollingConnection = false;
+      }
+
+      if (isLongPollingConnection !== null && !hasRaisedTimeToFirstByteEvent) {
+        hasRaisedTimeToFirstByteEvent = true;
+
+        if (onOpenTimestamp === null) {
```

Comment:
If you anticipate that this check never fires, then you should make it a `debugAssert`, which removes overhead from the production build. If you anticipate that it might crash, then you need to handle it in a way that doesn't cause the SDK to crash.

FWIW, I think this check breaks your code in 252, since it will be validated for a CONNECT_ATTEMPT, which fires before any other stat event.